src/mono/netcore/sample/iOS/Program.csproj

```
@@ -25,7 +25,7 @@
 
   <Import Project="/Users/egorbo/prj/runtime/tools-local/tasks/mobile.tasks/AotCompilerTask/MonoAOTCompiler.props" />
   <UsingTask TaskName="AppleAppBuilderTask"
-             AssemblyFile="$(ArtifactsBinDir)AppleAppBuilder\$(Configuration)\$(NetCoreAppCurrent)\AppleAppBuilder.dll" />
+             AssemblyFile="$(ArtifactsBinDir)AppleAppBuilder\$(Platform)\$(NetCoreAppCurrent)\AppleAppBuilder.dll" />
```

This will always be Debug since they're built as part of the tools-local tasks step.

---

@safern the path doesn't contain Configuration anymore as far as I understand, e.g. `/x64/net5.0` so it doesn't matter if it's debug or release

---

Since we moved them to tools-local tasks, it includes the configuration. i.e, from a CI build:

```
MonoAOTCompiler -> /Users/runner/runners/2.168.2/work/1/s/artifacts/bin/MonoAOTCompiler/Debug/net5.0/MonoAOTCompiler.dll
  AppleAppBuilder -> /Users/runner/runners/2.168.2/work/1/s/artifacts/bin/AppleAppBuilder/Debug/net5.0/AppleAppBuilder.dll
  installer.tasks -> /Users/runner/runners/2.168.2/work/1/s/artifacts/bin/installer.tasks/Debug/netstandard2.0/installer.tasks.dll
```

https://dev.azure.com/dnceng/public/_build/results?buildId=661296&view=logs&j=2ad48c51-3a9b-54d8-4f99-23e9592c4176&t=67200ffe-bea9-5ae3-c02c-88e6addf62e0&l=51

---

You can replace it with `$(AppleAppBuilderTasksAssemblyPath)`:
```suggestion
             AssemblyFile="$(AppleAppBuilderTasksAssemblyPath)" />
```

---

@safern ah, you are right, changed.

---

@akoeplinger it's evaluated to "" for some reason, no idea why root directory props is not used

---

I believe it is because the samples don't import the `Directory.Build.props` files from above.

---

Hm yeah we might want to fix this eventually so we don't need to duplicate all these properties.